core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -714,6 +715,16 @@ boolean adjustSplitBatchSize(List<SplitAssignment> splitAssignments, long reques
     }
 
     private void sendUpdate()
+    {
+        try {
+            sendUpdateUnsafe();
+        }
+        catch (Throwable e) {
```

Comment:
Can we actually do that holistically? Maybe catch specific section in `sendUpdate`. It looks like `sendUpdate` might be touching some state (e.g `updateErrorTracker#acquireRequestPermit`)

Reply:
IDK what you mean by that

Reply:
I looked at `sendUpdate` and it seems it's generally stateless.
I don't think the code after https://github.com/trinodb/trino/blob/5e5811065061202b09745e05eb4a5adde9764835/core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java#L777 could fail, which seems like the only plance where early method termination would break things.